packages/compat/src/v1-config.ts

```diff
@@ -42,7 +42,16 @@ export class WriteV1Config extends Plugin {
     if (this.storeConfigInMeta) {
       contents = metaLoader();
     } else {
-      contents = `export default ${JSON.stringify(this.inputTree.readConfig())};`;
+      contents = `
+      import { isTesting } from '@embroider/macros';
+      let env;
+      if (isTesting()) {
```

Comment:
Discussed in office hours: It would be cleaner to only include the `if (isTesting()) {` logic in the module contents if we have a `testInputTree`. eg. something like:

```ts
      if (this.testInputTree) {
        contents = `
        import { isTesting } from '@embroider/macros';
        let env;
        if (isTesting()) {
          env = ${JSON.stringify(this.testInputTree.readConfig())};
        } else {
          env = ${JSON.stringify(this.inputTree.readConfig())};
        }
        export default env;
        `;
      } else {
        contents = `export default ${JSON.stringify(this.inputTree.readConfig())};`;
      }
```

Reply:
Implemented in https://github.com/embroider-build/embroider/pull/1412/commits/18aa5045f39afc3490be55082943cb1c4abfd508